### PR TITLE
dfu: mcuboot: rename boot_swap_type> mcuboot_swap_type

### PR DESCRIPTION
**Pull request by nashif**
This function conflicts with a function of the same name in mcuboot. This happens when building USB DFU support into mcuboot.

DFU over USB uses image manager and mcuboot internals to manage images downloaded to the device.

### CODE REVIEW

**Comment by nvlsianpu** [comment]
> They conflict if we build the bootloader
with USB DFU.

What a bootloader? It is something else than mcuboot?

**Comment by nvlsianpu** [comment]
Honestly - this change is wired - better to change function names in application your attempt to build, or if this is relay impossibly - change names here.

**Comment by nashif** [comment]
> What a bootloader? It is something else than mcuboot?

yes, mcuboot.

**Comment by nashif** [comment]
> Honestly - this change is wired - better to change function names in application your attempt to build, or if this is relay impossibly - change names here.

dfu usb depends on image manager and pulls mcuboot.c, mcuboot.c duplicates function names in mcuboot, but those are only used when you have the application use mcumgr (which does not make sense to enable in the bootloader), so just use those duplicated functions with mcumgr and not when building mcuboot. I do not see anything weird about that.

**Comment by nvlsianpu** [comment]
Don't see such file name in mcuboot mainline - so this is something you working on.

USB DFU (DFU class instead of serial recovery?) in bootloader should use mcuboot primitives instead of the zephyr primitives.

**Comment by nashif** [comment]
> Don't see such file name in mcuboot mainline - so this is something you working on.
> 

boot_swap_type is both in Zephyr and in mcuboot.


> USB DFU (DFU class instead of serial recovery?) in bootloader should use mcuboot primitives instead of the zephyr primitives.

it is up to me to decide how I want to build my bootloader and what features I want to add in.

File: subsys/dfu/boot/mcuboot.c [hunk 1]
```diff
@@ -349,6 +353,7 @@ int boot_read_bank_header(u8_t area_id,
 	return 0;
 }
 
+#ifdef CONFIG_MCUMGR
```
Comment (mbolivar) [inline subsys/dfu/boot/mcuboot.c]: I don't understand. These are static and there are no functions by the same name anywhere else in the tree. Why is this necessary?

Comment (mbolivar) [reply subsys/dfu/boot/mcuboot.c]: Missed the thread on this.

Can we just rename them in this file to not conflict or something, since they're static? No need for extra ifdefing.

Comment (nashif) [reply subsys/dfu/boot/mcuboot.c]: yes, renaming is probably a better option, I thought there might be some purpose behind the same names. Btw, we have functions defined in the is file that are used no where in the tree. dead code basically, i.e `boot_read_bank_header`

Comment (nashif) [reply subsys/dfu/boot/mcuboot.c]: > since they're static?

btw, the main culprit is not static but if we ifdef it, the others will be become 'unused', hence why they all were ifdef'ed

Comment (nvlsianpu) [reply subsys/dfu/boot/mcuboot.c]: > yes, renaming is probably a better option. I thought there might be some purpose behind the same names.

That is my opinion as well, so will approve. Name were the same as function does the same as in mucboot (and had no better name before) - only that.

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: samples/subsys/usb/dfu/README.rst [hunk 2]
```diff
@@ -86,7 +86,7 @@ showing this output to the console:
   [MCUBOOT] [INF] boot_status_source: Image 0: magic=unset, copy_done=0xff, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Scratch: magic=unset, copy_done=0xe, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Boot source: slot 0
-  [MCUBOOT] [INF] boot_swap_type: Swap type: test
+  [MCUBOOT] [INF] mcuboot_swap_type: Swap type: test
```
Comment (mbolivar) [inline samples/subsys/usb/dfu/README.rst]: This should not have been changed; those log messages are from mcuboot itself

File: samples/subsys/usb/dfu/README.rst [hunk 3]
```diff
@@ -102,7 +102,7 @@ USB DFU sample, showing this output to the console:
   [MCUBOOT] [INF] boot_status_source: Image 0: magic=good, copy_done=0x1, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Scratch: magic=unset, copy_done=0xe, image_ok=0xff
   [MCUBOOT] [INF] boot_status_source: Boot source: none
-  [MCUBOOT] [INF] boot_swap_type: Swap type: revert
+  [MCUBOOT] [INF] mcuboot_swap_type: Swap type: revert
```
Comment (mbolivar) [inline samples/subsys/usb/dfu/README.rst]: same here

File: samples/subsys/usb/dfu/README.rst (hunk 2 above)
Comment (nvlsianpu) [reply samples/subsys/usb/dfu/README.rst]: fix covered in #13552

File: samples/subsys/usb/dfu/README.rst (hunk 3 above)
Comment (nvlsianpu) [reply samples/subsys/usb/dfu/README.rst]: fix covered in #13552
